pkg/registry/controller/rest.go

```diff
@@ -42,12 +42,19 @@ func (rcStrategy) NamespaceScoped() bool {
 	return true
 }
 
-// ResetBeforeCreate clears the status of a replication controller before creation.
-func (rcStrategy) ResetBeforeCreate(obj runtime.Object) {
+// PrepareForCreate clears the status of a replication controller before creation.
+func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (rcStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newController := obj.(*api.ReplicationController)
+	oldController := old.(*api.ReplicationController)
+	newController.Status = oldController.Status
```

Comment:
I *think* replication controller does not have a subresource endpoint for status update (#4909), which means it relies on this update to update the status as well. Your change prevents that from happening.